packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -136,7 +136,7 @@ class Checkbox extends Foundation<
      * This method should be called after render because it relies on element references.
      */
     private applyIndeterminateState(): void {
-         if (this.props.indeterminate && this.inputRef.current && this.state.checked) {
+         if (this.props.indeterminate && this.inputRef.current) {
```

Comment:
@Mrigank11 thanks for the PR! I think we'll also need some styles added as part of this PR. For instance, if an indeterminate checkbox *is* checked, we'll want to make sure that the indeterminate style is the visual treatment. Let me know if you need a rundown on where to find and make that change.

Reply:
That'll be helpful, please point me to the right place.

Reply:
In the repo, we have a package called `fast-components-styles-msft` where the checkbox code lives. You can make the changes in that file. You'll need to make your changes in the styles and then run `lerna bootstrap` in order to sync and symlink the packages.

From a "how" this should come together, I would suggest extracting the "indeterminate" styles into a new function so they can be applied in two places (you can see an example in the `hypertext` file). 

If you need any additional help or questions answered, join us on [discord](https://discord.gg/FcSNfg4) and we'll be happy to help you!

Reply:
Resolving as this is currently working as expected.

Reply:
@chrisdholt I originally created this component and decided on the behavior. After discussions with the team, we decided that the indeterminate prop would be valid only when checked since it is sort of a gray area, and you could really go either way. I'm ok with this change, but I wanted to give some context.

Reply:
I added some links in the issues attached here; ultimately indeterminate does not require checked. There are some great examples in the links provided to go deeper into why I think this change is necessary. We can't actually support all valid indeterminate scenarios if we bind it to a state. Indeterminate is simply a visual state and not a functional state, so I think binding to a functional state will prevent us from accomplishing our goals.

Reply:
Sure, I support this.